src/models/flight_control/FGLinearActuator.cpp

```diff
@@ -0,0 +1,252 @@
+            std::cerr << "FGLinear_Actuator Cicle_zero " << InputNodes[0]->GetNameWithSign() << std::endl;
+            return true;
+        }
+        
+        Output = 0.0;
+        
+        set = 1.0;
+        reset = 0.0;
+        
+        if (isSetProperty) set = setProperty;
+        if (isResetProperty) reset = resetProperty;
+            
+        if (set > 0 ) {
+            
+            double input_delta = Input - input_prec;
+            
+            if ((versus < 0.0 and input_delta > 0.0) or (versus  > 0.0 and input_delta < 0.0)) {
+                input_delta = 0.0;
+            }
+            input_prec = Input;
+            
+            if (input_delta != 0.0) {
+                if (input_delta <= (module * maxRate) and input_delta >= (module * minRate)) {
+                    if (input_delta > 0.0) {
+                        direction = 1;
+                        //cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " direction + " << direction << endl;
+                    } else if (input_delta < 0.0) {
+                        direction = -1;
+                        //cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " direction - " << direction << endl;
+                    }
+                }
+                
+                if (direction != 0) {
+                    if (input_delta >= (module * maxRate) or input_delta <= (module * minRate)) {
+                        if (direction > 0) {
+                            countSpin++;
+                            cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " countSpin++ " << countSpin << " maxRate " << maxRate << " minRate " << minRate << endl;
+                            direction = 0;
+                        } else if (direction < 0) {
+                            countSpin--;
+                            cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " countSpin-- " << countSpin << " maxRate " << maxRate << " minRate " << minRate << endl;
+                            direction = 0;
+                        }
+                    }
+                }
+                
+            }
+        } else {
+            Input = input_prec;
+        }
+        
+        if (reset > 0) {
+            Input = 0.0;
+            countSpin = 0.0;
+        }
+        
+        Output = gain * (Input + module*countSpin);
+        Lag();
+        Clip();
+        if (IsOutput) SetOutput();
+        
+        return true;
+    }
+    
+    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    void FGLinear_Actuator::Lag(void)
+    {
+        // "Output" on the right side of the "=" is the current frame input
+        // for this Lag filter
+        double input = Output;
+        
+        if (lag > 0.0) {
+            if (lag == lagPrevius) {
+                Output = ca * (input + previousLagInput) + previousLagOutput * cb;
```

Comment:
I am pretty sure that `previousLagInput` and `input_prec` are redundant and contain the same value at any time.